### DISCUSSION TITLE
Q: Is the integration with pyodide possible?

### DISCUSSION THREAD

**Discussion by carlkl** [lona-web-org/lona#192] [Ideas]
Is the integration of pyodide (wasm) inside a lona-web application possible? 
I'm thinking about a matplotlib canvas with widgets defined by lona-web.

**Comment by fscherf** [comment]
Hi @carlkl!

I never used pyodide or wasm before, but it should be possible to integrate with lona because both integrate with javascript and lona is vanilla javascript only in the Frontend.

Can you tell us more about your use case and what you want to achieve?

**Comment by carlkl** [comment]
Hi @fscherf, thank you for the feedback. 

I was thinking about using pyodide for small physics simulations running in the browser. I want to choose pyodide because I need the whole numpy / scipy / matplotlib machinery for that task.

There is a very nice example on the net (unrelated to my work) - just to give an an idea what I want to achive:

https://www.itp.uni-hannover.de/de/ag/weimer/lehre/ideal-gas-simulator/rapid-python-gui-development-with-html5-and-pyodide/
https://www.itp.uni-hannover.de/fileadmin/itp/ag/weimer/idealgas/idealgas.tar.gz
and
https://www.itp.uni-hannover.de/de/ag/weimer/lehre/   (three demos)

Maybe lona could help to make the GUI part more pythonic (see idealgas.tar.gz). On the other hand this is possibly too short thought as pyodide once load runs in the browser without any connection to the server.

pyodide itself has support to asyncio and jinha2. However, there is no support of aiohttp right now.

Cheers @carlkl

**Reply by fscherf** [reply]
I think that should be no problem. Lona is designed to run side by side with pretty much everything, so in worst case you use Lona for GUI and interaction and tell it to ignore the parts of your website that run pyodide. For an example on how to integrate frontend libraries into Lona, take a look at lona-chartjs or the documentation. If you encounter problems, just ask :)
